### PR TITLE
Remove multi-variable list initialization

### PR DESCRIPTION
**Pull request by zachschoenfeld33**
Disallow list-based variable declaration and initialization such as `int[32] x, y, z;`. 

Require instead that each variable be declared in its own statement, ie
```
int[32] x;
int[32] y;
int[32] z;
```

This applies to classical and quantum types. ANTLR, tests and examples are updated accordingly.

### CODE REVIEW

**Comment by zachschoenfeld33** [comment]
@steleman

**Comment by hhorii** [comment]
I believe that this is based on the discussion in https://github.com/Qiskit/openqasm/issues/141.

Maybe how close to C is a point to be discussed for this PR. https://github.com/Qiskit/openqasm/issues/131 said that "OpenQASM 3 is loosely based on C".

File: source/language/classical.rst [hunk 1]
```diff
@@ -47,8 +47,8 @@ right circular shift, ``rotl`` and ``rotr``, respectively.
 
 .. code-block:: c
 
-   bit[8] a = "10001111";
-   bit[8] b = "01110000";
+   bit a[8] = "10001111";
+   bit b[8] = "01110000";
```
Comment (blakejohnson) [inline source/language/classical.rst:51]: Unrelated change.

File: source/language/insts.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ broadcasts to ``b[j] = measure a[j];`` for each index ``j`` into register ``a``.
 
    // Initialize, flip and measure an array of 10 qubits
    qubit qubits[10];
-   bit[10] bits;
+   bit bits[10];
```
Comment (blakejohnson) [inline source/language/insts.rst:50]: Unrelated change.
